common/remote/rmtsmtp.cpp

```diff
@@ -487,10 +491,24 @@ class CMailInfo
             throw MakeStringException(0, "email sender address too long: %" I64F "u characters",  static_cast<__uint64>(sender.length()));
         validator.validateAddress(sender.get(), "email sender address");
 
-        getRecipients(validator, _to);
+        getRecipients(validator, _to, to);
         if(strlen(toHeader) + to.length() > 998)
             throw MakeStringException(0, "Email recipient address list too long: %u characters", to.length());
 
+        if (_cc && _cc[0])
+        {
+            getRecipients(validator, _cc, cc);
+            if(strlen(ccHeader) + cc.length() > 998)
+                throw MakeStringException(0, "Email cc: recipient address list too long: %u characters", cc.length());
+        }
+
+        if (_bcc && _bcc[0])
+        {
+            getRecipients(validator, _bcc, bcc);
+            if(strlen(bccHeader) + bcc.length() > 998)
```

Comment:
This 998 char limit seems a bit arbitrary (especially for bcc where people sometimes want 1000's of entries). Is that something from our code, of from the protocol?

Reply:
I copied that check to match what was already there, thinking there was a Good Reason for it.

I think someone read the RFC wrong and imposed that limit where it didn't need to be.  RFC 821 (SMTP) says that there is a 1000 character limit for a "text line" but that really refers to SMTP commands, not the body (which is where these recipient strings wind up).  I think we could probably remove this check; do you agree?

RFC 821 says that a maximum of 100 recipients (to + cc + bcc) is allowed, but that is a per-server config setting and, in the wild, the actual limit is all over the place.  This limit is not checked in our code, but I'm not sure if it is worthwhile to add it because the _real_ limit is imposed by the mail server.

Reply:
Perhaps leave as is for now but open another Jira to remove the limit should anyone have the time/need

Reply:
Sounds good.  I've squashed the commits.